packages/bot/src/db/models/Commit.ts

```diff
@@ -48,3 +48,48 @@ export async function getTopCommitsAfterDate(
     .orderBy("commit.committed_at", "desc")
     .limit(limitCount);
 }
+
+interface RepoViolationPercentage {
+  repoName: string;
+  percentage: string;
+}
+
+export async function getRepoViolationPercentages(): Promise<
+  RepoViolationPercentage[]
+> {
+  return db
+    .from<Commit>("commit as t1")
+    .select("repo_name as repoName")
+    .select(
+      db.raw(`ROUND((0.0 + (
```

Comment:
We usually avoid putting business logic (e.g. formula calculation) into SQL statements to keep them simple.

Instead, we can return the total count of commit and total count of commits with violation, and compute the percentage in server.